doc/releases/release-notes-4.1.rst

```diff
@@ -30,6 +30,11 @@ Removed APIs in this release
 Deprecated in this release
 ==========================
 
+* Deprecated the :c:func:`bt_le_set_auto_conn` API function. Application developers can achieve
+the same functionality in their application code by reconnecting to the peer when the
+:c:member:`bt_conn_cb.disconnected` callback is invoked.
```

Comment:
I am not sure if this should be here or under the Bluetooth Host section. Looking at previous release notes it seems that we put deprecation there. And it seems that some subsystem do the same.

@kartben do you have any recommendation on what we should do there?